src/System.Management.Automation/utils/PathUtils.cs

```diff
@@ -17,6 +18,42 @@ namespace System.Management.Automation
     /// </summary>
     internal static class PathUtils
     {
+
+        /// <summary>
+        /// THE method for opening a file for writing.
+        /// Should be used by all cmdlets that write to a file.
+        /// </summary>
+        /// <param name="cmdlet">cmdlet that is opening the file (used mainly for error reporting)</param>
+        /// <param name="filePath">path to the file (as specified on the command line - this method will resolve the path)</param>
+        /// <param name="encoding">encoding (this method will convert the command line string to an Encoding instance)</param>
+        /// <param name="defaultEncoding">if <c>true</c>, then we will use default .NET encoding instead of the encoding specified in <paramref name="encoding"/> parameter</param>
+        /// <param name="Append"></param>
+        /// <param name="Force"></param>
+        /// <param name="NoClobber"></param>
+        /// <param name="fileStream">Result1: <see cref="FileStream"/> opened for writing</param>
+        /// <param name="streamWriter">Result2: <see cref="StreamWriter"/> (inherits from <see cref="TextWriter"/>) opened for writing</param>
+        /// <param name="readOnlyFileInfo">Result3: file info that should be used to restore file attributes after done with the file (<c>null</c> is this is not needed)</param>
+        /// <param name="isLiteralPath">True if wildcard expansion should be bypassed.</param>
+        internal static void MasterStreamOpen(
```

Comment:
Suggest merging this with the Encoding overload. There appears to be only 1 caller to the later overload and it starts converts a FileEncoding to an Encoding to accomplish it.